articles/django-crud-api/index.md

```diff
@@ -0,0 +1,293 @@
+Requirement already satisfied: pytz in ./venv/lib/python3.8/site-packages (from django>=2.2->djangorestframework) (2020.4)
+Installing collected packages: djangorestframework
+Successfully installed djangorestframework-3.12.2
+```
+
+Lets create out django-todo project now by running the command `django-admin startproject django_todo`
+```python
+$ django-admin startproject django_todo
+```
+After successfully creating the project we change our working directory to our project folder and run the Django development server.
+```python
+$ cd django_todo
+$ ./manage.py runserver
+Watching for file changes with StatReloader
+Performing system checks...
+
+System check identified no issues (0 silenced).
+
+You have 18 unapplied migration(s). Your project may not work properly until you apply the migrations for app(s): admin, auth, contenttypes, sessions.
+Run 'python manage.py migrate' to apply them.
+November 17, 2020 - 10:14:01
+Django version 3.1.3, using settings 'django_todo.settings'
+Starting development server at http://127.0.0.1:8000/
+Quit the server with CONTROL-C.
+```
+
+When we visit http://127.0.0.1:8000/ we should see the Django welcome page. Django makes it easy for us to organize our code into reusable apps. We are going to create an app for the todo endpoints.
+
+To create an app, run:
+
+```python
+$ python3 manage.py startapp todo
+```
+The above command creates a directory `todo` and generates boilerplate code for a todo app. Now we can plug our todo app into our `django_todo` project.
+
+Open `settings.py` in the project directory `django_todo` and add our `todo` app to the `INSTALLED_APPS` list. We also add the `rest_framework` app to make it available for use in our project.
+```python
+# ./django_todo/settings.py
+...
+INSTALLED_APPS = [
+    ...
+    'django.contrib.messages',
+    'django.contrib.staticfiles',
+    'todo',
+    'rest_framework',
+]
+```
+### Creating the Todo model
+
+Models are python classes that get mapped to the database tables in the database. The Todo Model will be mapped to the todo table in the database. All Django models subclass the model. Model superclass.
+In the `models.py` in the `todo` app, we will create our `Todo` model will the below code.
+
+```python
+from django.db import models
+
+# Create your models here.
+class Todo(models.Model):
+    title = models.CharField(max_length = 100)
+    body = models.CharField(max_length = 100)
+    is_completed = models.BooleanField(default=False)
+    date_created = models.DateField(auto_created=True)
+    last_modified = models.DateField(auto_now=True)
+
+    def __str___(self):
+        return self.title 
+```
+To create an SQLite database and Todo table in the database, we will run the command `./manage.py makemigrations` to create SQL queries that will be used to create a `Todo` table in the database. To create the Todo table in the database will run the command `./manage.py migrate`.
```

Comment:
The idea around migrations is a little confusing, even for me. Consider pointing the reader to the [migration docs](https://docs.djangoproject.com/en/3.1/topics/migrations/).
Then rewrite these as sequential steps like so:

To make migrations run:
`./manage.py makemigrations`

To apply the migrations run:
`./manage.py migrate`